invenio_communities/records/records/systemfields/communities/field.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# Invenio-Communities is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Communities system field."""
+
+from invenio_records.systemfields import SystemField
+
+from .context import CommunitiesFieldContext
+from .manager import CommunitiesRelationManager
+
+class CommunitiesField(SystemField):
+    """Communites system field for manaing relations to communities."""
+
+    def __init__(self, m2m_model_cls, key='communities', context_cls=None,
+                 manager_cls=None):
+        self._m2m_model_cls = m2m_model_cls
+        self._context_cls = context_cls or CommunitiesFieldContext
+        self._manager_cls = manager_cls or CommunitiesRelationManager
+        super().__init__(key=key)
+
+    #
+    # Life-cycle hooks
+    #
+    def pre_commit(self, record):
+        """Commit the communities field."""
+        manager = self.obj(record)
+        self.set_dictkey(record, manager.to_dict())
+
+    #
+    # Helpers
+    #
+    def obj(self, record):
+        """Get or crate the communities manager."""
+        # Check cache
+        obj = self._get_cache(record)
+        if obj is not None:
```

Comment:
It took me an hour to track down a bug related to this line 🥸 A classic one of ``if obj`` vs ``if obj is None`` - even though we got an object from the cache, it sometimes evaluated to false because we implemented ``__len__()``

Reply:
at some point we should probably check the code-base for such potential bugs :detective: 

one occasion where they're actually warranted is when checking `LocalProxy`'d objects – in this case, the `… is None` check will never evaluate to `True`, even if the object behind the proxy is set to `None` (after all, the checked reference is still the proxy)
in such cases, the `not …` will still work however, since i think it does an implicity `bool(…)` cast and thus resolves the object behind the proxy
example: https://github.com/inveniosoftware/invenio-userprofiles/blob/master/invenio_userprofiles/api.py#L30 where the `current_user` proxy is used